File: strategy/python/example1/src/libs/ducks/mallardduck.py
from libs.ducks.duck import Duck
from libs.behaviors.fly.flywithwings import FlyWithWings
from libs.behaviors.quack.quack import Quack

class MallardDuck(Duck): 
    flyBehavior = FlyWithWings()
    quackbehavior = Quack()

    def display(self):
        print("I am a mallard duck!")


    

File: strategy/typescript/example1/libs/duck/mallard-duck.ts
import { FlyWithWings } from "../behaviors/fly/fly-with-wings";
import { Quack } from "../behaviors/quack/quack";
import { Duck } from "./duck";

export class MallardDuck extends Duck {
  protected flyBehaviour = new FlyWithWings();
  protected quackBehaviour = new Quack();

  public display(): void {
    console.log("I'm a real mallard duck");
  }
}
